### PR TITLE
Add update database to client

### PR DESCRIPTION
**Pull request by kiranpandit**
Adding support for update database: https://developers.notion.com/reference/update-a-database

### CODE REVIEW

File: src/api-endpoints.ts [hunk 1]
```diff
@@ -277,6 +278,37 @@ export const databasesCreate = {
...
+interface DatabasesUpdateBodyParameters {
+  properties?: UpdatePropertySchemaMap
+  title?: RichTextInput[]
+}
+
+export interface DatabasesUpdateParameters
+  extends DatabasesUpdatePathParameters,
+    DatabasesUpdateQueryParameters,
+    DatabasesUpdateBodyParameters {}
+export interface DatabasesUpdateResponse extends BlockBase {}
```
Comment (almawang) [inline src/api-endpoints.ts:302]: This returns a `Database` object right? not `BlockBase`?

File: src/api-types.ts [hunk 2]
```diff
@@ -1040,3 +1040,28 @@ export interface LastEditedTimePropertySchema {
 export interface LastEditedByPropertySchema {
```
Comment (almawang) [inline src/api-types.ts:1040]: This is not part of the PR, but why is this a `PropertySchema`? this shouldn't be editable right?

File: src/api-types.ts [hunk 3]
```diff
@@ -1040,3 +1040,28 @@ export interface LastEditedTimePropertySchema {
...
+
+type UpdateSelectOptionSchema = SelectOptionSchema | SelectOption
+
+type UpdateMultiSelectOptionSchema = UpdateSelectOptionSchema
+export interface UpdateSelectPropertySchema {
+  select: { options?: UpdateSelectOptionSchema[] }
+}
+
+export interface UpdateMultiSelectPropertySchema {
+  multi_select: { options?: UpdateMultiSelectOptionSchema[] }
```
Comment (almawang) [inline src/api-types.ts:1059]: Why have a separate type for the mulitselect options?

File: src/api-types.ts (hunk 2 above)
Comment (almawang) [reply src/api-types.ts:1040]: Same for `  | CreatedTimePropertySchema
  | CreatedByPropertySchema
  | LastEditedTimePropertySchema
  | LastEditedByPropertySchema`

**Review by almawang — CHANGES_REQUESTED** [review]
I think the `Response` type needs to be updated and remove the non updatable property schema types

File: src/api-types.ts (hunk 2 above)
Comment (kiranpandit) [reply src/api-types.ts:1040]: these can be added to the database schema as properties, the values themselves are not updated

File: src/api-types.ts (hunk 3 above)
Comment (kiranpandit) [reply src/api-types.ts]: good point - updated so both use the same type

File: src/api-endpoints.ts (hunk 1 above)
Comment (kiranpandit) [reply src/api-endpoints.ts]: good catch, fixed!

File: src/api-types.ts (hunk 2 above)
Comment (almawang) [reply src/api-types.ts:1040]: ah i see, sg!

File: src/api-types.ts [hunk 4]
```diff
@@ -1040,3 +1040,27 @@ export interface LastEditedTimePropertySchema {
...
+ * Update database property schema (input)
+ */
+export interface RenamePropertySchema {
+  name: string
+}
+
+type UpdateSelectOptionSchema = SelectOptionSchema | SelectOption
+
+export interface UpdateSelectPropertySchema {
+  select: { options?: UpdateSelectOptionSchema[] }
```
Comment (almawang) [inline src/api-types.ts:1054]: the `options` field is nullable? what's the behavior when an empty object is passed into `select`?

**Review by almawang — APPROVED** [review]
LGTM, question about the behavior of an empty `select` field

File: src/api-types.ts (hunk 4 above)
Comment (kiranpandit) [reply src/api-types.ts:1054]: if it's undefined, then there will be no options (similar to when you create a "select" property in the client before adding any options). In [create database](https://developers.notion.com/reference/create-a-database#select-configuration), the `options` are optional (ha).
